docker.md

```diff
@@ -0,0 +1,184 @@
+
+## Running Gem with Docker
+
+This document explains how to deploy Gem inside docker on localhost. A next step is getting this tooling to deploy and manage remote machines, but it doesn't do that yet.
+
+### About Docker
+
+Docker ("docker engine") is a system service that manages **containers**. A container is a program that runs with its own Linux environment, but all containers on a machine share the same kernel. So it's kind of like a VM but it's better to think of as a unit of isolation for applications.
+
+Containers are based on **images** which are immutable filesystem layers that define an application's environment. The application itself is added on top of an existing base image like `ubuntu` or `centos` for instance. These are publicly available.
+
+The running state of our system is two docker containers, one running Gem (just telnetd for now) and one running Postgres. The Postgres container uses the `postgres:9.6.0` image which is maintained by Postgres. The Gem container uses the `java:latest` base image with our application added on via sbt-native-packager, yielding (for now) an image called `telnetd:0.1-SNAPSHOT`.
```

Comment:
Right, it does all that for you. You can customize it as much as you like but the default works so I haven't messed with it that much.